stubs/CoreGenericFunctions.phpstub

```diff
@@ -627,12 +627,23 @@ function substr_replace($string, $replace, $offset, $length = null) {}
 
 /**
  * @psalm-pure
- *
  * @param string $haystack
- *
+ * @param string|int $needle
```

Comment:
Seems to contradict the type on the param below

Reply:
I've removed the int type now. It was for this

> needle
Prior to PHP 8.0.0, if needle is not a string, it is converted to an integer and applied as the ordinal value of a character. This behavior is deprecated as of PHP 7.3.0, and relying on it is highly discouraged. Depending on the intended behavior, the needle should either be explicitly cast to string, or an explicit call to chr() should be performed.

Seeing as Psalm doesn't support 7.3 I think it's safe to ignore it?

Reply:
Psalm doesn't run on PHP 7.3 but it can still analyze 7.1+ code anyway (it can work on older code too but there may be method signatures mismatches)

However, given the discouragement/deprecation/change for this, I think we should consider this as an error even for previous versions and only support string

Reply:
You'll have to move the test that fails because of this to invalid tests case to ensure this throws an error now